app/utils/AuthUtils.scala

```diff
@@ -16,32 +16,39 @@
 
 package utils
 
+import models.AtedSubscriptionAuthData
 import play.api.Logger
-import uk.gov.hmrc.play.frontend.auth.AuthContext
-import uk.gov.hmrc.play.frontend.auth.connectors.domain.{AgentAdmin, AgentAssistant}
+import uk.gov.hmrc.auth.core.{Admin, AffinityGroup, Assistant, User}
 
 object AuthUtils {
 
-  def isAgent(implicit user: AuthContext): Boolean = user.principal.accounts.agent.isDefined
+  def isAgent(implicit user: AtedSubscriptionAuthData): Boolean =
+    user.enrolments.getEnrolment("HMRC-AGENT-AGENT").isDefined || user.affinityGroup.contains(AffinityGroup.Agent)
 
-  def isAgentAssistant(implicit user: AuthContext): Boolean = user.principal.accounts.agent.map(_.agentUserRole).contains(AgentAssistant)
+  def isAgentAssistant(implicit user: AtedSubscriptionAuthData): Boolean = user.credentialRole.contains(Assistant)
 
-  def isAgentAdmin(implicit user: AuthContext): Boolean = user.principal.accounts.agent.map(_.agentUserRole).contains(AgentAdmin)
+  def isAgentAdmin(implicit user: AtedSubscriptionAuthData): Boolean =
+    user.credentialRole.contains(Admin) || isAgent && user.credentialRole.contains(User)
 
-  def agentLink(implicit user: AuthContext): String = {
-    user.principal.accounts.agent.map(_.link).getOrElse {
+  def agentLink(implicit user: AtedSubscriptionAuthData): String = {
+    user.agentCode.map(str => s"/agent/$str").getOrElse {
       Logger.warn(s"[AuthUtils][getAgentLink] Exception - User does not have the correct authorisation ")
       throw new RuntimeException("User is not agent")
```

Comment:
Unhandled exception when no agent code is found.

Reply:
Preserving old functionality